src/mtev_conf.c

```diff
@@ -1315,6 +1315,9 @@ mtev_conf_section_to_xpath(mtev_conf_section_t* section) {
     } else {
       buff_len = current_entry_len + sizeof("/descendant::[999999]\0");
       current_entry = malloc(buff_len);
+      if (buff_len != 512) {
+        free(buff);
+      }
       buff = current_entry;
```

Comment:
I think this is ugly and fragile.  Could we instead have a `char *onheap = NULL;` that is assigned in here as `onheap = buff = current_entry;` and free(onheap)?